sdcm/cluster_aws.py

```diff
@@ -185,14 +186,36 @@ def _create_instances(self, count, ec2_user_data='', dc_idx=0):
                        'SubnetId': self._ec2_subnet_id[dc_idx],
                        'AssociatePublicIpAddress': True,
                        'Groups': self._ec2_security_group_ids[dc_idx]}]
-
-        if self.instance_provision == INSTANCE_PROVISION_ON_DEMAND:
+        if self.instance_provision == 'mixed':
+            instances = self._create_mixed_instances(count, interfaces, ec2_user_data, dc_idx)
+        elif self.instance_provision == INSTANCE_PROVISION_ON_DEMAND:
             instances = self._create_on_demand_instances(count, interfaces, ec2_user_data, dc_idx)
         else:
             instances = self._create_spot_instances(count, interfaces, ec2_user_data, dc_idx)
 
         return instances
 
+    def _create_mixed_instances(self, count, interfaces, ec2_user_data, dc_idx):
+        instances = []
+        if isinstance(self, ScyllaAWSCluster) or isinstance(self, CassandraAWSCluster):
+            if count > 2:
+                count_on_demand = 2
+            else:
+                count_on_demand = 1
```

Comment:
Quoted from commitlog: `- db seeds nodes are created as SPOT instances, not seeds nodes as On demand.`

So the `count_on_demand` should equal to   count - `seeds_num`?


Here we also need to consider the case of adding node/nodes to existing cluster, such as grow_cluster or terminate_replace nemesis. The counter is 1, but we expected to additionally add a new db node (non-seed), so it should be sopt.

Reply:
OK , then the commit log needs to be updated, or you can add this request by commentss

In terminate_and_replace_node, we need to add a new spot or demand instance depends on which kind of instance was removed.

And if we call add_nodes(count=1,....), then it always create on_demand instance. It's problem.

Reply:
New node/nodes which are added by add_nodes will be added as SPOT instances if instance_provision in yaml will be mixed. Because after cluster will be initiated first time, then the self.instance_provision will be set to INSTANCE_PROVISION_SPOT_LOW_PRICE.
Here we will have a question, what do we expect? @roydahan @bentsi : 
1. is ok? if was terminated on_demand instance in mixed mode, then instead of it will be created the spot instance
2. or we should some how detect which node was removed, and create new one with same type?

Reply:
@amoskong is right, I think we should add the same type as we terminated.
Otherwise, you can reach a state all non_seed_nodes will be spot.

Reply:
implemented next logic:
number of on_demand instances couldn't be more than 2.
if upon adding new instance instead of terminamted, the number of on_demand instance <2, then add  on_demand instace,
if number of on_demand instances >= 2, then adding spot instances only

Reply:
ok